src/api/clinical/diagnosis/diagnosis.controller.ts

```diff
@@ -91,13 +91,12 @@ export class DiagnosisController extends BaseController{
         try {
 
             const domainModel = await this._validator.update(request);
-
             const id: uuid = await this._validator.getParamUuid(request, 'id');
             const existingUser = await this._service.getById(id);
             if (existingUser == null) {
                 throw new ApiError(404, 'Diagnosis record not found.');
             }
-            await this.authorizeUser(request, existingUser.MedicalPractitionerUserId);
+            await this.authorizeOne(request, existingUser.Patient.UserId, existingUser.Patient.TenantId);
             const updatedDiagnosis = await this._service.update(id, domainModel);
```

Comment:
This needs to be looked at. There are some problems.

Reply:
Diagnosis dto was not returning patient user id. Modifyied the mapper in order to return the patient user id.